app/models/user.rb

```diff
@@ -28,7 +20,7 @@ def set_location
   # Include default devise modules. Others available are:
   # :omniauthable
   devise :database_authenticatable, :registerable, :confirmable, :timeoutable,
-         :recoverable, :rememberable, :trackable, :validatable, :lockable
```

Comment:
I explain it it the commit message.

"""
Unify email and password validations

They were split between Devise internal code and the User model, making
them hard to understand and customize. I put them all together in the
model.
"""

I hardly ever use the Devise `validatable` module because I hardly ever want the exact validations [devise provides](https://github.com/plataformatec/devise/blob/88724e10adaf9ffd1d8dbfbaadda2b9d40de756a/lib/devise/models/validatable.rb#L29-L35). In this case, we were using them but also using specific email validations, so it's hard to know which one is actually applied or which one we actually want, and also doing unnecessary extra validation work. I prefer to keep all validations together in the model and assume it's not devise's job to perform validations on model attributes.

Regarding the comment you propose, what I usually do is remove the configuration blocks of the devise modules I'm not using. Do you want me to do that, to avoid confusion?